Sources/Player.swift

```diff
@@ -441,7 +458,7 @@ extension Player {
         let keys: [String] = [PlayerTracksKey, PlayerPlayableKey, PlayerDurationKey]
 
         self._asset.loadValuesAsynchronously(forKeys: keys, completionHandler: { () -> Void in
-            DispatchQueue.main.sync(execute: { () -> Void in
+            DispatchQueue.main.async(execute: { () -> Void in
```

Comment:
@piemonte I totally get that. I may have misunderstood the `sync` vs `async` but my understanding was that it was saying "I need the main thread now" (sync) vs "I need the main thread ASAP" (async) and since I was getting to that point before viewDidLoad finished, I was getting a crash. 

However like I said, your guard to make sure we are loaded before getting in there should also fix the issue so I'm happy to switch back to `sync` if you want!